airbyte-integrations/connectors/source-exchange-rate-api/source_exchange_rate_api/source.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+import datetime
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream, IncrementalMixin
+from airbyte_cdk.sources.streams.http import HttpStream
+
+from source_exchange_rate_api.authenticator import HTTPApiKeyHeaderAuth
+
+
+class ExchangeRateApiStream(HttpStream, ABC):
+    url_base = "https://api.apilayer.com/exchangerates_data/"
+
+    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
+        super(ExchangeRateApiStream, self).__init__(*args, **kwargs)
+        self.config: Mapping[str, Any] = config
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield response.json()
+
+
+class LatestRates(ExchangeRateApiStream):
+    primary_key = "timestamp"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "latest"
+
+    def request_params(
```

Comment:
shouldn't it be better to move this method to the base class so it could be reused?

Reply:
you mean stream? now we have two streams and from what I see their params are identical